server/methods/core/groups.app-test.js

```diff
@@ -0,0 +1,135 @@
+    }
+
+    expect(createGroup).to.throw(Meteor.Error, /Access Denied/);
+  });
+
+  it("should add a user to a group successfully and reference the id on the user account", function () {
+    sandbox.stub(Reaction, "hasPermission", () => true);
+    spyOnMethod("createGroup", shop._id);
+    spyOnMethod("addUser", shop._id);
+
+    Meteor.call("group/createGroup", sampleGroup, shop._id);
+    const group = Groups.findOne({ shopId: shop._id });
+    Meteor.call("group/addUser", user._id, group._id);
+    const updatedUser = Accounts.findOne({ _id: user._id });
+    expect(updatedUser.groups).to.include.members([group._id]);
+  });
+
+  it("should add a user to a group and update user's permissions", function () {
+    sandbox.stub(Reaction, "hasPermission", () => true);
+    spyOnMethod("createGroup", shop._id);
+    spyOnMethod("addUser", shop._id);
+
+    Meteor.call("group/createGroup", sampleGroup, shop._id);
+    const group = Groups.findOne({ shopId: shop._id });
+    Meteor.call("group/addUser", user._id, group._id);
+    const updatedUser = Meteor.users.findOne({ _id: user._id });
+
+    expect(updatedUser.roles[shop._id]).to.include.members(sampleGroup.permissions);
+  });
+
+  it("should remove a user from a group and update user's permissions", function () {
+    sandbox.stub(Reaction, "hasPermission", () => true);
+    spyOnMethod("createGroup", shop._id);
+    spyOnMethod("addUser", shop._id);
+    spyOnMethod("removeUser", shop._id);
+
+    Meteor.call("group/createGroup", sampleGroup, shop._id);
+    const group = Groups.findOne({ shopId: shop._id });
+    Meteor.call("group/addUser", user._id, group._id);
+    let updatedUser = Meteor.users.findOne({ _id: user._id });
+    expect(updatedUser.roles[shop._id]).to.include.members(sampleGroup.permissions);
+
+    Meteor.call("group/removeUser", user._id, group._id);
+    updatedUser = Meteor.users.findOne({ _id: user._id });
+    expect(updatedUser.roles[shop._id]).to.not.include.members(sampleGroup.permissions);
+  });
+
+  it("should ensure a user's permissions get updated when the group permissions changes", function () {
+    sandbox.stub(Reaction, "hasPermission", () => true);
+    spyOnMethod("createGroup", shop._id);
+    spyOnMethod("addUser", shop._id);
+    spyOnMethod("updateGroup", shop._id);
+
+    Meteor.call("group/createGroup", sampleGroup, shop._id);
+    const group = Groups.findOne({ shopId: shop._id });
+    Meteor.call("group/addUser", user._id, group._id);
+    let updatedUser = Meteor.users.findOne({ _id: user._id });
+    expect(updatedUser.roles[shop._id]).to.include.members(sampleGroup.permissions);
+
+    const newGroupData = Object.assign({}, sampleGroup, { permissions: ["new-permissions"] });
+    Meteor.call("group/updateGroup", group._id, newGroupData, shop._id);
+    updatedUser = Meteor.users.findOne({ _id: user._id });
+    expect(updatedUser.roles[shop._id]).to.include.members(newGroupData.permissions);
```

Comment:
Might be worth adding a test to make sure the permissions that the old group had, but the new group does _not_ have are no longer included.